### ISSUE TITLE
[Bug] Hex search finds what it wasn't asked for

### ISSUE THREAD

**Issue by 4aiman** [WerWolv/ImHex#461]
### Operating System

Windows

### What's the issue you encountered?

HEX search finds stuff it's not supposed to.
![image](https://user-images.githubusercontent.com/2913210/156346509-50bf35f2-0732-473f-b7d0-c2a5d4d66905.png)

### How can the issue be reproduced?

Open a file
Go file-Search
Switch to HEX tab
Search your HEX string

In my case I've searched for 5C5C and got matches with 2C5C, 735C, 305C, 725C, etc

### ImHex Version

1.16.1

### ImHex Build Type

- [ ] Nightly or built from sources

**Comment by WerWolv** [comment]
Weird. I can't seem to replicate this at all. Even searching for `5C5C` in a file that has that combination as well as others yields the correct results for me.

How big is your file? Does it span over multiple pages perhaps?

**Comment by 4aiman** [comment]
The document is a single-line JSON about 73Mb in size. Single page. Sadly, I'm not allowed to upload it here.
5C is only an example.
Every HEX search is affected
![image](https://user-images.githubusercontent.com/2913210/156550564-6bf7e2a6-5fae-4055-8fbd-dc8f5a8be224.png)

As for 2 pages or more...
I used this pattern, copied&pasted it several times to get over 300Mb and saved.
![image](https://user-images.githubusercontent.com/2913210/156550666-a75ed05a-5a8c-4423-b31a-ee227c34db2d.png)
Then I opened that file with ImHex and tried to search for ```2c5c```
Imhex just sat there doing nothing. CPU load didn't increase, GPU load didn't increase.
